### ISSUE TITLE
React-native tests are broken after installation of react-native-code-push

### ISSUE THREAD

**Issue by AndreyMitsyk** [Microsoft/react-native-code-push#1245]
**Re-pro steps:**
1) Create new ReactNative project: ```react-native init TestProj``` and navigate to project directory
2) Run default test and make sure that test completed without errors: ```npm test```
3) Install CodePush 
```npm install --save react-native-code-push```
```npm install```
```react-native link react-native-code-push```
4) Enable code-push inside app.js:
```
...
var codePush = require("react-native-code-push");
...
class App extends Component<{}> {
...
App = codePush(App);
export default App;
```
5) Verify default react native tests: ```npm test```

**Expected:** Default test executed without errors.

**Actual:** 
```
> jest

 FAIL  __tests__\App.js
  ● Test suite failed to run

    TypeError: codePush is not a function

    > 1 | /**
      2 |  * Sample React Native App
      3 |  * https://github.com/facebook/react-native
      4 |  * @flow

      at Object.<anonymous> (App.js:1:2844)
      at Object.<anonymous> (__tests__/App.js:1:194)
```

**Comment by iTOYS** [comment]
There is typings issue(`react-native-code-push.d.ts`)

**Comment by AndreyMitsyk** [comment]
Workaround to not use codePush function inside tests:
Add initialization of CodePush into index.js instead of app.js.

```
import { AppRegistry } from 'react-native';
import App from './App';
import codePush from "react-native-code-push";

var root = codePush(App);
AppRegistry.registerComponent('NewTestApp', () => App);
```
Only initialization (codePush(App)) should be added inside index.js. Other logic like codePush.sync can be added inside app.js.  This will help in some scenarios.

**Comment by slaci** [comment]
I went with the mock solution, but I think the `cp` variable should be changed to this: `const cp = options => (typeof options === 'function' ? options : app => app);` at least for codepush 6.

This is because you call it in two ways:
* `codePush({opt1: 'val'})(rootComponent)`
* `codePush(rootComponent)`

I tested it with both class and function root components and it works for me both ways. Maybe someone find it useful.
